client/src/js/actions/groupAction.js

```diff
@@ -0,0 +1,306 @@
+ * @return {*} empty
+ */
+export function removeGroupMember(userId) {
+  return {
+    type: REMOVE_GROUP_MEMBER_SUCCESS,
+    userId
+  };
+}
+
+export const fetchUserGroups = () => (dispatch) => {
+  dispatch(getGroup());
+  // making call to the get all groups API endpoint
+  return Api(null, '/api/v1/groups/', 'GET').then(
+    (userGroups) => {
+      dispatch(getGroupSuccess(userGroups));
+    }
+  ).catch((groupError) => {
+    dispatch(getGroupFail(groupError));
+  });
+};
+
+export const createGroup = (name, desc) => (dispatch) => {
+  const groupCreateParams = `name=${name}&desc=${desc}`;
+  // making calls to the create group API endpoint
+  return Api(groupCreateParams, '/api/v1/groups/', 'POST').then(
+    (createGroupResponse) => {
+      const group = createGroupResponse.group;
+      // setting current group to store
+      dispatch(setCurrentGroup(group));
+      return group;
+    }
+  ).catch((addError) => {
+    dispatch(addGroupFail(addError));
+  });
+};
+
+
+  // onSearchUserInGroup Method
+export const onSearchUser = (id, searchText) => dispatch => Api(
+  null,
+  `/api/v1/groups/${id}/usersearch?search=${searchText}`,
+  'GET')
+  .then(
+    userSearchResult => userSearchResult.searchItemResult
+  );
+
+  // TODO: fix users in a group and deleting user from group UI & backend
+export const setSelectedGroupMembers = groupId => (dispatch) => {
+  // TODO: make API call to get users of a group from the server
+  return Api(null, `/api/v1/groups/${groupId}/users/`, 'GET').then(
+    (currentGroupMembers) => {
+      // setting current group members response from server to store
+      dispatch(setCurrentGroupUsersSuccess(currentGroupMembers));
+      return currentGroupMembers;
+    }
+  ).catch(
+    (groupMembersError) => {
+      // setting error response from server when getting members of group
+      dispatch(setCurrentGroupUsersFail(groupMembersError));
+      return groupMembersError;
+    });
+};
+
+  // onAddUserToGroup Method
+export const onAddUser = (userId, groupId) => (dispatch) => {
+  const addUserParams = `userId=${userId}`;
+  return Api(addUserParams, `/api/v1/groups/${groupId}/user/`, 'POST').then(
+    (addUserToGroupResponse) => {
+      // check if response, i.e user added, set message state to "user added"
+      // check for users in a group and update UI to add mark icon
+      return addUserToGroupResponse;
+    }
+
+  );
+};
+
+// onAddUserToGroup Method
+export const onRemoveUser = (userId, groupId) => (dispatch) => {
+  const removeUserParams = `?usersId=${userId}`;
+  return Api(null, `/api/v1/groups/${groupId}/user/${removeUserParams}`, 'DELETE').then(
+    (removeUserFromGroupResponse) => {
+      console.log('removeUserFromGroupResponse========>', removeUserFromGroupResponse);
+      if (removeUserFromGroupResponse.message === 'User Is not an Admin') {
+        return removeUserFromGroupResponse;
+      }
+      if (removeUserFromGroupResponse.message === 'You cannot remove yourself') {
```

Comment:
Both if statements on lines 247 and 250 can be merged into one block since *removeUserFromGroupResponse* is returned in both instances.